README.md

```diff
@@ -22,79 +30,90 @@ Also, add the applinks:<APPSITE HOST> entry to the Associated Domains Capability
-console.log(gatheredTokens);
-
-// Prints:
-//
-// { access_token: '...', refresh_token: '...', id_token: '...', ...}
-// OR
-// undefined
+KeyCloak.login(conf, callback, scope)
+  .then((response) => /* Your resolve */ );
+  .catch((error) => /* Your reject*/ )
+```
+Method arguments:
+  - _conf_: The JSON configuration object (see the example below).
+  - _callback_: By default the plugin try to open the keycloak login url on the default browser. Using this callback you can override this behavior e.g. handling the login flow into a WebView without leaving the app.
+  - _scope_: By default its value is 'info'. You can override this argument if some custom KeyCloak behavior is needed (e.g if you need to handle the KeyCloak ID_TOKEN, you have to pass 'openid info offline_access' as value).
+
+```json
+config = {
+  "realm": "<real_name>",
+  "auth-server-url": "https://<domain>/sso/auth/",
+  "ssl-required": "string",
+  "resource": "<resource_name>",
+  "credentials": {
+    "secret": "<secret_uuid>"
+  },
+  "confidential-port": "string",
+}
 ```
 
-### Login
-```js
-
-const config = {
-  url: 'https://<KEYCLOAK_HOST>/auth',
-  realm: '<REALM NAME>',
-  resource: '<RESOURCE NAME>',
-  redirectUri: 'https://<REDIRECT HOST>/success.html',
-  appsiteUri: 'https://<APPSITE HOST>/app.html',
-  kcIdpHint: 'facebook', // *optional*
-};
-
-Login.startLoginProcess(config).then(tokens => {
-  console.log(tokens);
-});
-
-// Prints:
-//
-// { access_token: '...', refresh_token: '...', id_token: '...', ...}
+Resolver arguments:
+ - _response_: a JSON object containing two fields:
+    - *tokens*: a JSON containing all the tokens returned by KeyCloak. If you used'info' as *scope* the JSON will be as shown below.
+    - *deepLinkUrl*: The redirectUrl with some KeyCloak query params added at the end.
+
+```json
+response.tokens = {
+    "access_token": "string",
+    "expires_in": "number",
+    "refresh_expires_in": "number",
+    "refresh_token": "string",
+    "token_type": "string",
+    "not-before-policy": "number",
+    "session_state": "string",
+    "scope": "string",
+}
 ```
 
-Logging in by the startLoginProcess function will save it in the AsyncStorage, whereas after its been successful, getTokens will get the most recent tokens that are saved and you can then use it to authenticate against a backend.
+#### Manually handling the tokens
 
-### Refreshing the token
 ```js
-const refreshedTokens = await Login.refreshToken();
-console.log(refreshTokens);
-// Prints:
-//
-// { access_token: '...', refresh_token: '...', id_token: '...', ...}
-// OR
-// undefined
+import KeyCloak, { TokenStorage } from 'react-native-keycloak-plugin'
 ```
 
+Logging in by the login function will save the tokens information into the AsyncStorage. Through the TokenStoage object, the plugin export three methods that can be used to get, save and clear the tokens. This method are needed if you want to directly access and manage the tokens from the AsyncStorge.
```

Comment:
Typo: "TokenStorage" , not "TokenStoage" 
Grammar: "The plugin exports three methods" (s for 3rd person) 
Typo: "These methods are", not "This method are"